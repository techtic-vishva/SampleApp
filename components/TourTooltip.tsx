import * as React from 'react'
import { View, TouchableOpacity, StyleSheet } from 'react-native'
import { dichotomousHippopotamus, fill } from '../constants/Colors'
import { TooltipProps } from 'rn-tourguide'
import { MainText, HeaderText } from './StyledText'

const Tooltip = ({
	isFirstStep,
	isLastStep,
	handleNext,
	handlePrev,
	handleStop,
	currentStep,
	labels
}: TooltipProps) => {
	const [p1, p2] = (currentStep?.text || '').split(':TITLE:')
	const title = p2 ? p1 : ''
	const body = title ? p2 : p1
	return (
		<View style={styles.container}>
			<View style={styles.desContainer}>
				{title && <HeaderText style={{ color: fill, marginBottom: 20, fontSize: 16 }}>{title}</HeaderText>}
				<MainText testID="stepDescription" style={styles.desText}>
					{body}
				</MainText>
			</View>
			<View style={styles.btnContainer}>
				{!isLastStep ? (
					<TouchableOpacity style={{ flex: 1 }} onPress={handleStop}>
						<MainText style={styles.btnText}>{labels?.skip || 'Skip'}</MainText>
					</TouchableOpacity>
				) : null}
				{!isFirstStep ? (
					<TouchableOpacity style={{ flex: 1 }} onPress={handlePrev}>
						<MainText style={styles.btnText}>{labels?.previous || 'Previous'}</MainText>
					</TouchableOpacity>
				) : null}
				{!isLastStep ? (
					<TouchableOpacity style={{ flex: 1 }} onPress={handleNext}>
						<MainText style={styles.btnText}>{labels?.next || 'Next'}</MainText>
					</TouchableOpacity>
				) : (
					<TouchableOpacity style={{ flex: 1 }} onPress={handleStop}>
						<MainText style={styles.btnText}>{labels?.finish || 'Finish'}</MainText>
					</TouchableOpacity>
				)}
			</View>
		</View>
	)
}

const styles = StyleSheet.create({
	container: {
		borderRadius: 16,
		paddingTop: 24,
		alignItems: 'center',
		justifyContent: 'center',
		paddingBottom: 16,
		width: '90%',
		backgroundColor: '#ffffffef'
	},
	desText: {
		fontFamily: 'objektiv',
		textAlign: 'center',
		color: fill
	},
	desContainer: {
		flex: 1,
		alignItems: 'center',
		justifyContent: 'space-around',
		width: '80%',
		paddingBottom: 10
	},
	btnText: {
		fontFamily: 'objektiv',
		textAlign: 'center',
		color: dichotomousHippopotamus
	},
	btnContainer: {
		marginTop: 10,
		flexDirection: 'row',
		justifyContent: 'flex-end',
		alignItems: 'center'
	}
})
export default Tooltip
