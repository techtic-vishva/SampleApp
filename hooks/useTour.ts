import { useIsFocused } from '@react-navigation/native'
import { useEffect } from 'react'
import { useTourGuideController } from 'rn-tourguide'
import { readString, storeString } from '../core/Storage'

type tours = 'home' | 'group'

function buildStorageKey(tourKey: tours) {
	return `TOUR:V1:${tourKey}`
}

export async function hasSeenTour(tourKey: tours) {
	const storageKey = buildStorageKey(tourKey)
	const tourSeen = !!(await readString(storageKey))
	return tourSeen
}

export default function useTour(tourKey: tours, autoStart?: boolean | (() => boolean)) {
	const storageKey = buildStorageKey(tourKey)
	const shouldAutoStart =
		typeof autoStart === 'boolean' ? autoStart : typeof autoStart === 'function' ? autoStart() : false

	const { start, canStart, TourGuideZone, eventEmitter } = useTourGuideController(tourKey)

	// Early bail for non-render locations
	if (!shouldAutoStart) return { canStart, start, TourGuideZone }

	// Setup render hooks
	const isFocused = useIsFocused()

	async function onCanStart() {
		if (!canStart || !isFocused) return

		const tourSeen = await hasSeenTour(tourKey)
		if (tourSeen) {
			return
		}

		start()
	}

	useEffect(() => {
		onCanStart().catch(() => {})
	}, [canStart, isFocused])

	useEffect(() => {
		const handleStop = async () => {
			storeString(storageKey, 'true')
		}

		eventEmitter?.on('stop', handleStop)
		return () => eventEmitter?.off('stop', handleStop)
	}, [canStart, eventEmitter])

	return {
		canStart,
		start,
		TourGuideZone
	}
}
