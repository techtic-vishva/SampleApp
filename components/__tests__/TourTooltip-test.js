import * as React from 'react'
import renderer from 'react-test-renderer'
import TourTooltip from '../TourTooltip'
import useTour from '../../hooks/useTour'
it('TourTooltip renders correctly', () => {
	const tree = renderer
		.create(
			<TourTooltip
				isFirstStep={false}
				isLastStep={true}
				handleNext={() => {}}
				handlePrev={() => {}}
				handleStop={() => {}}
				currentStep={{ text: 'test', order: 1, target: '', wrapper: '' }}
				labels={{ skip: 'skip', previous: 'previous', next: 'next', finish: 'finish' }}
			/>
		)
		.toJSON()
	expect(tree).toMatchSnapshot()
})
