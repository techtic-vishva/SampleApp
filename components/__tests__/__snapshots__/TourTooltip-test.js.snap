// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TourTooltip renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ffffffef",
      "borderRadius": 16,
      "justifyContent": "center",
      "paddingBottom": 16,
      "paddingTop": 24,
      "width": "90%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "space-around",
        "paddingBottom": 10,
        "width": "80%",
      }
    }
  >
    
    <Text
      style={
        Array [
          Object {
            "color": "#fff",
            "fontFamily": "objektiv",
          },
          Array [
            Object {
              "color": "#23211e",
              "fontFamily": "objektiv",
              "textAlign": "center",
            },
            Object {
              "fontFamily": "objektiv",
            },
          ],
        ]
      }
      testID="stepDescription"
    >
      test
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "flex-end",
        "marginTop": 10,
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flex": 1,
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontFamily": "objektiv",
            },
            Array [
              Object {
                "color": "#ed832d",
                "fontFamily": "objektiv",
                "textAlign": "center",
              },
              Object {
                "fontFamily": "objektiv",
              },
            ],
          ]
        }
      >
        previous
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flex": 1,
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontFamily": "objektiv",
            },
            Array [
              Object {
                "color": "#ed832d",
                "fontFamily": "objektiv",
                "textAlign": "center",
              },
              Object {
                "fontFamily": "objektiv",
              },
            ],
          ]
        }
      >
        finish
      </Text>
    </View>
  </View>
</View>
`;
